c7n/policy.py

```diff
@@ -73,8 +75,13 @@ def __init__(self, data, options):
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            if 'all' in options.regions:
+                options.regions = utils.get_available_regions(p['resource'])
```

Comment:
i'm a bit torn, we can do this expansion at the same place we lookup the default region in the cli and where we'll be doing the account id, instead of moving it down here. ie we don't have to create transient sessions in multiple places just for config if we just use one for all three usages.